foreshadow/steps/cleaner.py

```diff
@@ -9,6 +9,50 @@
 from .preparerstep import PreparerStep
 
 
+def _check_empty_columns(X) -> NoReturn:
+    """Check if all columns are empty in the dataframe.
+
+    Args:
+        X: the dataframe
+
+    Returns:
+        the empty columns.
+
+    Raises:
+        ValueError: all columns are dropped.
+
+    """
+    columns = pd.Series(X.columns)
+    empty_columns = columns[X.isnull().all(axis=0).values]
+
+    if len(empty_columns) == len(columns):
+        error_message = (
+            "All columns are dropped since they all have "
+            "over 90% of missing values. Aborting foreshadow."
+        )
+        logging.error(error_message)
+        raise ValueError(error_message)
+    elif len(empty_columns) > 0:
+        logging.info(
+            "Dropping columns due to missing values over 90%: {}"
```

Comment:
Please make this log more general in case we have other reasons to drop columns. We can also provide more information in the data cleaner function that decides to drop the column.

Reply:
Done. The message has been updated as "Identified columns with over 90% missing values."